File: src/unity3d/kumakore_sample/Assets/com.kumakore.test/TestInventory.cs
using System;
using com.kumakore;
using NUnit.Framework;
using System.Threading;

namespace com.kumakore.test
{
    [TestFixture]
    public class TestInventory : TestBase
	{
		// valid item name
		private static readonly String ITEM_NAME = "com_sinuouscode_red_ball";
		
        public static void All()
        {
            TestInventory inventory = new TestInventory();
            inventory.setup();
			
			inventory.SyncAddItem();
			inventory.SyncRemoveItem();
			inventory.AsyncAddItem();
			inventory.AsyncRemoveItem();
			
        }
		
		[TestFixtureSetUp]
		public override void setup ()
		{
			base.setup ();
			app1 ().signin(TEST_1,PASSWORD).sync ();
		}
		
		
		[Test]
        public void SyncAddItem()
        {
			int quantity = 1;
            app1().getUser().getInventory().addItem(ITEM_NAME,quantity).sync(delegate(ActionInventoryAdd action)
            {
                Assert.AreEqual(StatusCodes.SUCCESS, action.getCode());
				Assert.GreaterOrEqual(app1 ().getUser ().getInventory()[ITEM_NAME].getQuantity(),quantity);
            });
        }

        [Test]
        public void AsyncAddItem()
        {
			int quantity = 2;
            app1().getUser().getInventory().addItem(ITEM_NAME,quantity).async(delegate(ActionInventoryAdd action)
            {
                Assert.AreEqual(StatusCodes.SUCCESS, action.getCode());
				Assert.GreaterOrEqual(app1 ().getUser ().getInventory()[ITEM_NAME].getQuantity(),quantity);
                Release();
            });

            Wait();
        }
		
		[Test]
        public void SyncRemoveItem()
        {
			int quantity = 0;
			if(app1().getUser ().getInventory().ContainsKey(ITEM_NAME)) quantity = app1 ().getUser().getInventory()[ITEM_NAME].getQuantity();
			else Assert.Pass (ITEM_NAME + " not found in inventory");
            app1().getUser().getInventory().removeItem(ITEM_NAME,quantity).sync(delegate(ActionInventoryRemove action)
            {
                Assert.AreEqual(StatusCodes.SUCCESS, action.getCode());
				Assert.True(!app1 ().getUser().getInventory().ContainsKey(ITEM_NAME) || app1 ().getUser().getInventory()[ITEM_NAME].getQuantity() == 0); 
            });
        }

        [Test]
        public void AsyncRemoveItem()
        {
            int quantity = 0;
			if(app1().getUser ().getInventory().ContainsKey(ITEM_NAME)) quantity = app1 ().getUser().getInventory()[ITEM_NAME].getQuantity();
            else Assert.Pass (ITEM_NAME + " not found in inventory");
            app1().getUser().getInventory().removeItem(ITEM_NAME,quantity).async(delegate(ActionInventoryRemove action)
            {
                Assert.AreEqual(StatusCodes.SUCCESS, action.getCode());
				Assert.True(!app1 ().getUser().getInventory().ContainsKey(ITEM_NAME) || app1 ().getUser().getInventory()[ITEM_NAME].getQuantity() == 0);
                Release();
            });

            Wait();
        }
	}
}

File: src/android/kumakore_sample/src/com/kumakore/sample/test/TestInventory.java
package com.kumakore.sample.test;

import android.util.Log;

import com.kumakore.ActionInventoryAdd;
import com.kumakore.ActionInventoryGet;
import com.kumakore.ActionInventoryRemove;
import com.kumakore.ItemBundle;
import com.kumakore.KumakoreApp;

public class TestInventory extends TestBase {
	private static final String TAG = TestInventory.class.getName();

	public TestInventory(KumakoreApp app) {
		super(app);
	}
	
	@Override
	protected void onRun() {
		testInventoryGet();
		testInventoryAdd();
		testInventoryRemove();			
	}
	
	public void testInventoryGet() {
		app().getUser().inventory().get().async(new ActionInventoryGet.IKumakore() {

			@Override
			public void onActionInventoryGet(ActionInventoryGet action) {
				// Log.i(TAG, "Inventory size " +
				// app.getUser().inventory().size());
				for (ItemBundle item : app().getUser().inventory()) {
					Log.i(TAG, item.getName() + "  " + item.getQuantity());
				}
			}
		});
	}

	public void testInventoryAdd() {

		app().getUser().inventory().addItem("amazing_powerup", 1).async(new ActionInventoryAdd.IKumakore() {

			@Override
			public void onActionInventoryAdd(ActionInventoryAdd action) {
				Log.i(TAG, "Inventory size " + app().getUser().inventory().size());
				for (ItemBundle item : app().getUser().inventory()) {
					Log.i(TAG, item.getName() + "  " + item.getQuantity());
				}

			}
		});
	}

	public void testInventoryRemove() {

		app().getUser().inventory().removeItem("amazing_powerup", 1).async(new ActionInventoryRemove.IKumakore() {

			@Override
			public void onActionInventoryRemove(ActionInventoryRemove action) {
				Log.i(TAG, "Inventory size " + app().getUser().inventory().size());
				for (ItemBundle item : app().getUser().inventory()) {
					Log.i(TAG, item.getName() + "  " + item.getQuantity());
				}

			}
		});

	}
}
